---
title: "Rbosome"
output: rmarkdown::pdf_document
vignette: >
  %\VignetteIndexEntry{Rbosome}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

## Installing Rbosome package
```{r}
devtools::install_github('rforbiodatascience22/group_3_package')
```

## Loading Rbosome package
```{r setup}
library(Rbosome)
```

## Package's functionality

Rbosome package delivers an easy way to work with the central dogma of molecular biology.

It provides basic functions to:

* create a random DNA strand

* transform DNA to RNA

* transform RNA to a list of codons

* translate RNA codons to letter code of amino acids

* generate an amino acids' abundance plot

-------------
## Functions' usage - example

### Create a DNA string consisting of 90 nucleotides:
```{r}
set.seed(26)
DNA_seq <- ATGC(90)
DNA_seq
```

\newpage
### Transform DNA string to RNA
```{r}
RNA_seq <- DNA_to_RNA(DNA_seq)
RNA_seq
```

### Make a list of RNA_string's codons
```{r}
codons <- rna_transform(RNA_seq)
codons
```

### Translate codons to amino acids
```{r}
AA_seq = translate(codons)
AA_seq
```

### Create abundance plot
```{r, fig.width=7, fig.height=5}
aa_abundance_plot(AA_seq)
```



